src/BuildScriptGenerator/DotNetCore/DefaultAspNetCoreWebAppProjectFileProvider.cs

```diff
@@ -132,19 +133,42 @@ public string GetProjectFile(ISourceRepo sourceRepo)
         // To enable unit testing
         internal static bool IsAspNetCoreWebApplicationProject(XDocument projectFileDoc)
         {
-            var webSdkProjectElement = projectFileDoc.XPathSelectElement(
-                DotnetCoreConstants.WebSdkProjectXPathExpression);
-            return webSdkProjectElement != null;
+            // For reference
+            // https://docs.microsoft.com/en-us/visualstudio/msbuild/project-element-msbuild?view=vs-2019
+
+            // Look for the attribute value on Project element first as that is more common
+            // Example: <Project Sdk="Microsoft.NET.Sdk.Web">
+            var expectedWebSdkName = DotnetCoreConstants.WebSdkName.ToLowerInvariant();
+            var sdkAttributeValue = projectFileDoc.XPathEvaluate(
+                DotnetCoreConstants.ProjectSdkAttributeValueXPathExpression);
+            var sdkName = sdkAttributeValue as string;
+            if (!string.IsNullOrEmpty(sdkName))
+            {
+                return sdkName.ToLowerInvariant().StartsWith(expectedWebSdkName);
```

Comment:
If this is false, shouldn't we let it proceed to the next check, with the `Sdk` node?
Thinking of this case:
```
<Project Sdk="bla">
   <Sdk Name="Microsoft.NET.Sdk.Web... />
...
```

Reply:
Sure, sounds good